pkg/resolver/client/ens/ens.go

```diff
@@ -121,12 +141,46 @@ func (c *Client) Close() error {
 	return nil
 }
 
-func wrapResolve(backend bind.ContractBackend, name string) (string, error) {
+func wrapDial(endpoint string, contractAddr string) (*ethclient.Client, *goens.Registry, error) {
 
-	// Connect to the ENS resolver for the provided name.
-	ensR, err := goens.NewResolver(backend, name)
+	// Dial the eth client.
+	ethCl, err := ethclient.Dial(endpoint)
 	if err != nil {
-		return "", err
+		return nil, nil, fmt.Errorf("wrapdial: dial: %w", err)
+	}
+
+	// Obtain the ENS registry.
+	registry, err := goens.NewRegistryAt(ethCl, common.HexToAddress(contractAddr))
+	if err != nil {
+		return nil, nil, fmt.Errorf("wrapdial: new registry: %w", err)
+	}
+
+	// Ensure that the ENS registry client is deployed to the given contract address.
+	_, err = registry.Owner("")
+	if err != nil {
+		return nil, nil, fmt.Errorf("wrapdial: owner: %w", err)
+	}
+
+	return ethCl, registry, nil
+}
+
+func wrapResolve(registry *goens.Registry, addr common.Address, name string) (string, error) {
```

Comment:
The same for this function, all `wrapresolve: ` can be removed from error messages. Also just a suggestion.

Reply:
true. i guess i went too far with the annotations :) thanks @janos